src/stores/thumbnails.js

```diff
@@ -0,0 +1,143 @@
+import { defineStore } from 'pinia'
+import { LRUCache } from 'lru-cache'
+import { deleteCachedImageFromBasename, getImageFromBasename, getAllCacheKeys } from '@/utils/imageLoader'
+
+
+const cacheOptionsSmall = {
+  max: 5000,
+  dispose: (value, key, reason) => {
+    if (reason != 'set'){
+      deleteCachedImageFromBasename('small', key)
+    }
+  },
+  noUpdateTTL: true,
+  // cache files for one week
+  ttl: 1000 * 60 * 60 * 24 * 7,
+  ttlResolution: 100*60*60,
+  allowStale: false,
+  updateAgeOnGet: true,
+  updateAgeOnHas: true,
+}
+
+const cacheOptionsLarge = {
+  max: 1000,
+  dispose: (value, key, reason) => {
+    if (reason != 'set'){
+      deleteCachedImageFromBasename('large', key)
+    }
+  },
+  noUpdateTTL: true,
+  // cache files for one week
+  ttl: 1000 * 60 * 60 * 24 * 7,
+  ttlResolution: 100*60*60,
+  allowStale: false,
+  updateAgeOnGet: true,
+  updateAgeOnHas: true,
+}
+
+export const useThumbnailsStore = defineStore('thumbnails', {
+  state() {
+    return {
+      archives: [],
+      smallThumbnailsCache: new LRUCache(cacheOptionsSmall),
+      largeThumbnailsCache: new LRUCache(cacheOptionsLarge),
+    }
+  },
+  persist: {
+    key: 'thumbnails',
+    serializer: {
+      serialize: value => {
+        let serializedValue = {
+          archives: value.archives,
+          smallThumbnailsCache: value.smallThumbnailsCache.dump(),
+          largeThumbnailsCache: value.largeThumbnailsCache.dump()
+        }
+        return JSON.stringify(serializedValue)
+      },
+      deserialize: value => {
```

Comment:
I think a simple comment here like:

```
// De-serialize image caches from persistent storage, invalidate object URLs, and refresh LRU caches on browser reload
```

would just help us when we're looking at this code in a years' time